src/map/utils/synthutils.cpp

```diff
@@ -835,6 +836,29 @@ int32 startSynth(CCharEntity* PChar)
 int32 doSynthResult(CCharEntity* PChar)
 {
     uint8 m_synthResult = PChar->CraftContainer->getQuantity(0);
+    if (map_config.anticheat_enabled)
+    {
+        std::chrono::duration animationDuration = server_clock::now() - PChar->m_LastSynthTime;
+        if (animationDuration < 5s)
+        {
```

Comment:
Is this not [already being covered in map/packet_system.cpp](https://github.com/DarkstarProject/darkstar/blob/master/src/map/packet_system.cpp#L3776-L3780)?

Reply:
No, apparently it only checks how long after you finish synth before you can start synth again, it doesn't check if you inject a synth end packet right after beginning.
I've verified that by using the said "fastcraft" addon. Without the patch I was able to craft without waiting for the animation.